src/Async/README.md

```diff
@@ -0,0 +1,1369 @@
+# Async
+
+```haskell
+Async e a = Rejected e | Resolved a
+```
+
+Defined as a "lazy" Sum Type that implements an asynchronous control structure,
+`Async` represents either the success or failure of a given asynchronous
+operation. The "laziness" of `Async` allows the ability to build complex
+asynchronous operations by defining each portion of that flow as smaller "steps"
+that can be composed together.
+
+Depending on your needs, an `Async` can be constructed in a variety of ways. The
+typical closely resembles how a `Promise` is constructed with one major
+difference, the arguments used in the function that is passed to the `Promise`
+constructor are reversed in an `Async`.
```

Comment:
It may not be the right place to explain it in the docs. But I can do it here: So it is typical and easier to follow when the far right parameter of a type is the covariant functor (the portion that works with `map`). So  because of this, the `Resolved` is the parameter that works with `map`. So to keep everything lined up and easier to follow, the arguments to the functions are in the same order.

Reply:
That said I think I can throw something in like:
> ... reversed in an `Async` *to match the order in which `Async` is parameterized*.

Do you think that could provide some hint to why?

Reply:
Yes, it can help IMHO, but it's your call here.